### PR TITLE
Add CosmosDB track 2 SDK and one scenario test

### PR DESCRIPTION
**Pull request by YalinLi0312**

### CODE REVIEW

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/CosmosDBManagementClientBase.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        internal CosmosDBManagementClient GetCosmosDBManagementClient()
+        {
+            return CreateClient<CosmosDBManagementClient>(this.SubscriptionId,
+                TestEnvironment.Credential,
+                Recording.InstrumentClientOptions(new CosmosDBManagementClientOptions()));
+        }
+
+        public async Task<string> GetLocation()
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/CosmosDBManagementClientBase.cs:43]: Is this method used? If not, remove.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/CosmosDBManagementClientBase.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        protected CosmosDBManagementClientBase(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        protected void InitializeClients()
+        {
+            SubscriptionId = TestEnvironment.SubscriptionId;
+            ResourcesManagementClient = GetResourceManagementClient();
+            ResourcesOperations = ResourcesManagementClient.Resources;
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/CosmosDBManagementClientBase.cs:30]: I don't think this is used anywhere.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/Helpers/CosmosDBTestUtilities.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+using Azure.Management.Resources;
+using Azure.Management.Resources.Models;
+using System.Threading.Tasks;
+
+namespace Azure.ResourceManager.CosmosDB.Tests
+{
+    public static class CosmosDBTestUtilities
+    {
+        internal const string ResourceGroupPrefix = "DefaultCosmosDB";
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/Helpers/CosmosDBTestUtilities.cs:12]: Use the recoding method like this to get a "unique" id: Recording.GenerateAssetName("Default-CosmosDB-");

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/OperationsTest.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        public async Task CleanupResourceGroup()
+        {
+            await CleanupResourceGroupsAsync();
+        }
+
+        [TestCase]
+        public async Task ListOperationsTest()
+        {
+            var cosmosDBClient = GetCosmosDBManagementClient();
+            var operations = cosmosDBClient.Operations.ListAsync();
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/OperationsTest.cs:37]: should await here as to make sure this operation completes before the assert.

**Review by bquantump — CHANGES_REQUESTED** [review]
Please resolve comments.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/Helpers/CosmosDBTestUtilities.cs (hunk 3 above)
Comment (YalinLi0312) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/Helpers/CosmosDBTestUtilities.cs:12]: Here is the prefix to generate unique id, it is using the recording method in test file:
https://github.com/Azure/azure-sdk-for-net/blob/9e997cfdd7fc79b5355a0d7a4ebebac341bba931/sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/OperationsTest.cs#L24

**Comment by allenjzhang** [comment]
Let's get this in asap. It is failing other comosdb track 1 PR checks. Thanks.

**Review by bquantump — APPROVED** [review]

**Pull request merged**